source/gateway_httproute_test.go

```diff
@@ -1506,10 +1505,8 @@ func TestGatewayHTTPRouteSourceEndpoints(t *testing.T) {
 			src, err := NewGatewayHTTPRouteSource(clients, &tt.config)
 			require.NoError(t, err, "failed to create Gateway HTTPRoute Source")
 
-			var b *bytes.Buffer
-			if len(tt.logExpectations) > 0 {
-				b = testutils.LogsToBuffer(log.DebugLevel, t)
-			}
```

Comment:
:thinking: What led you to remove the condition here ?

Reply:
As initial issue seems like is resolved. I'm saying `seems` as data races the hardest to resovle. The `-race` flag is a race detector, this condition was to trick `-race` detector

Reply:
I could keep it, no harm